src/main/java/seedu/contax/logic/parser/RangeCommandParser.java

```diff
@@ -0,0 +1,101 @@
+
+
+/**
+ * Parses input arguments and creates a new RangeCommandParser object
+ */
+public class RangeCommandParser {
+    /**
+     * Parses the given {@code String} of arguments in the context of the RangeEditCommandParser
+     * and returns an RangeEditCommandParser object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RangeCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_RANGE_FROM, PREFIX_RANGE_TO,
+                        PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+
+        Index fromIndex;
+        Index toIndex;
+        if (argMultimap.getValue(PREFIX_RANGE_FROM).isPresent()) {
+            fromIndex = Index.fromOneBased(Integer.parseInt(argMultimap.getValue(PREFIX_RANGE_FROM).get()));
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RangeCommand.MESSAGE_EDIT_USAGE));
+        }
+        if (argMultimap.getValue(PREFIX_RANGE_TO).isPresent()) {
+            toIndex = Index.fromOneBased(Integer.parseInt(argMultimap.getValue(PREFIX_RANGE_TO).get()));
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RangeCommand.MESSAGE_EDIT_USAGE));
+        }
+
+        switch (argMultimap.getPreamble()) {
+
+        case "edit":
+            EditCommand.EditPersonDescriptor editPersonDescriptor = new EditCommand.EditPersonDescriptor();
+
+            if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, RangeCommand.MESSAGE_EDIT_USAGE));
+            }
+            if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+                editPersonDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()));
+            }
+            if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+                editPersonDescriptor.setEmail(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()));
+            }
+            if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+                editPersonDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
+            }
+            parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(editPersonDescriptor::setTags);
+
+            if (!editPersonDescriptor.isAnyFieldEdited()) {
+                throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
+            }
+
+            return new RangeCommand(fromIndex, toIndex, editPersonDescriptor);
+
+        case "delete":
+            return new RangeCommand(fromIndex, toIndex, null);
+
+        default:
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RangeCommand.MESSAGE_USAGE));
+        }
```

Comment:
I feel there was no need to do this, you can just hand over to AddressBookParser and let the individual command parsers handle the parsing, you just need to do a loop to call it on multiple contacts.

Reply:
thx for the suggestion, changed accrodingly